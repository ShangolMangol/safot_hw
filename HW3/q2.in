fun sum a b c = a + b + c;

structure SumKernel1D = Kernel1D(struct
	type source = int
	type target = int
	val kernel = sum
	fun default _ = 0
end);
structure SumKernel2D = Kernel2D(struct
	type source = int
	type target = int
	fun kernel (x1, x2, x3) (y1, y2, y3) (z1, z2, z3) = 2 * x2 + y2 + 3 * z2
	fun default _ = 0
end);

print "===TEST START===\n";
val test1 = if (SumKernel1D.runKernel [1, 2, 3, 4]) = [3, 6, 9, 7] then "PASSED" else "FAILED";
val test2 = if (zip [1, 2, 3] [4, 5, 6] [7, 8, 9]) = [(1, 4, 7), (2, 5, 8), (3, 6, 9)] then "PASSED" else "FAILED";
val test3 = if (fill false 3) = [false, false, false] then "PASSED" else "FAILED";
val test4 = if (SumKernel2D.runKernel [[1, 2, 3], [4, 5, 6], [7, 8, 9]]) = [[7, 13, 7], [19, 31, 16], [31, 49, 25]] then "PASSED" else "FAILED";