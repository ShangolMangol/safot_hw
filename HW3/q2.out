Standard ML of New Jersey (64-bit) v110.99.2 [built: Sun Mar 20 20:10:23 2022]
[opening hw3_q2.sml]
[opening hw3_q2_def.sml]
signature KERNEL1D_SIG =
  sig
  type source
  type target
  val kernel : source -> source -> source -> target
  val default : source -> source
end
signature KERNEL2D_SIG =
  sig
  type source
  type target
  val kernel :
      source * source * source
       -> source * source * source -> source * source * source -> target
  val default : source -> source
end
val it = () : unit
functor Kernel1D(Elt: sig
  type source
  type target
  val kernel : source -> source -> source -> target
  val default : source -> source
end) :
sig
  val runKernel : Elt.source list -> Elt.target list
end
val sum = fn : int -> int -> int -> int
structure SumKernel : sig
  val runKernel : Elt.source list -> Elt.target list
end
val it = [3,6,9,7] : int list
exception SizeNotEqual
val zip = fn : 'a list -> 'b list -> 'c list -> ('a * 'b * 'c) list
val it = [(1,4,7),(2,5,8),(3,6,9)] : (int * int * int) list
exception NegativeLength
val fill = fn : 'a -> int -> 'a list
val it = [false,false,false] : bool list
[autoloading]
[library $SMLNJ-BASIS/basis.cm is stable]
[library $SMLNJ-BASIS/(basis.cm):basis-common.cm is stable]
[autoloading done]
functor Kernel2D(Elt: sig
  type source
  type target
  val kernel :
      source * source * source
       -> source * source * source -> source * source * source -> target
  val default : source -> source
end) :
sig
  val runKernel : source list -> target list list
end
- val sum = fn : int -> int -> int -> int
structure SumKernel1D : sig
  val runKernel : Elt.source list -> Elt.target list
end
structure SumKernel2D : sig
  val runKernel : source list -> target list list
end
===TEST START===
val it = () : unit
val test1 = "PASSED" : string
val test2 = "PASSED" : string
val test3 = "PASSED" : string
val test4 = "PASSED" : string

