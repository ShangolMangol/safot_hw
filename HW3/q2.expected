val it = () : unit
val test1 = "PASSED" : string
val test2 = "PASSED" : string
val test3 = "PASSED" : string
val test4 = "PASSED" : string

